stustapay/core/schema/db/0000-base.sql

```diff
@@ -246,7 +253,10 @@ create table if not exists product (
     -- todo: ean or something for receipt?
     name text not null unique,
     -- price including tax (what is charged in the end)
-    price numeric not null,
+    price numeric,
+    -- price is not fixed, e.g for top up. Then price=null and set with the api call
+    fixed_price boolean not null default true,
+    constraint product_not_fixed_or_price check ( price is not null = fixed_price),
```

Comment:
`price_set = price is not null`

we want either price to be set, or a fixed price, not both. so `price_set` must be different from `fixed_price`

the constraint should hence be `(price is not null) != fixed_price`

Reply:
This constraint is already correct. We could change it to `price is null <> fixed_price` which has the same meaning. See the following examples are correct usage:
- fixed_price = True, price = 3.00
- fixed_price = False, price = null